tests/home.spec.ts

```diff
@@ -36,14 +40,15 @@ test.describe('Homepage', () => {
     {
       tag: ['@smoke', '@desktop', '@mobile'],
     },
-    async () => {
+    async ({ page }) => {
       await test.step('When user search with a valid keyword', async () => {
         await homePage.fillAndSearch('depapp');
       });
 
       await test.step('Then table should show expected result', async () => {
         await expect(homePage.getUsername('depapp')).toBeVisible();
         await expect(homePage.emptyState).not.toBeVisible();
+        await expect(page).toHaveScreenshot('search-results-valid.png', { maxDiffPixelRatio: 0.02 });
       });
```

Comment:
Is it better to add another step to assert the visual?

Something like this:
```suggestion
     await test.step('Then validate visual for search result', async () => {
        await expect(page).toHaveScreenshot('search-results-valid.png', { maxDiffPixelRatio: 0.02 });
     ]})
```

cc: @depapp